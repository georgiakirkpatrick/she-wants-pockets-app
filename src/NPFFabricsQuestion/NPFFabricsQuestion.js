import React from 'react'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPage from '../FormPage/FormPage'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const NPFFabricsQuestion = props => {
    const primCheckOption = [
        {
            text: 'Yes',
            id: 'primary-fabric',
            checked: props.primCheck
        }
    ]
    
    const secCheckOption = [
        {
            text: 'Yes',
            id: 'secondary-fabric',
            checked: props.secCheck
        }
    ]
    
    const linCheckOption = [
        {
            text: 'Yes',
            id: 'lining-fabric',
            checked: props.linCheck
        }
    ]

    return (
        <div id='fabric-question'>
            <FormPage title='Fabrics'>
                <FormPromptWithSub 
                    prompt='The primary fabric (or material) is the main fabric used in the garment or accessory.'
                />

                <FormCheckboxSection 
                    prompt='The primary fabric (or material) is required.'
                    options={primCheckOption}
                    selectedOptions={{'primary-fabric': props.primCheck}}
                    handleChange={() => props.setPrimCheck(props.primCheck)}
                />

                <FormPromptWithSub 
                    prompt='Garments and accessories sometimes include secondary fabrics as trims or accents.'
                />

                <FormCheckboxSection 
                    prompt='Does this product include a secondary fabric?'
                    options={secCheckOption}
                    selectedOptions={{'secondary-fabric': props.secCheck}}
                    handleChange={() => props.setSecCheck(!props.secCheck)}
                />

                <FormPromptWithSub 
                    prompt='Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.'
                />

                <FormCheckboxSection 
                    prompt='Does this product include a lining fabric?'
                    options={linCheckOption}
                    selectedOptions={{'lining-fabric': props.linCheck}}
                    handleChange={() => props.setLinCheck(!props.linCheck)}
                />
            </FormPage>
            <NPFFooter buttons='prevNext' previousButton={() => props.setPage(props.currentPage - 1)} nextButton={() => props.setPage(props.currentPage + 1)} />
        </div>
    )    
}

NPFFabricsQuestion.defaultProps = {
    
}

export default NPFFabricsQuestion