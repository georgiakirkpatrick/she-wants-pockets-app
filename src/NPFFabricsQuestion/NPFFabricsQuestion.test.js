import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NPFFabricsQuestion from './NPFFabricsQuestion'

describe('NPFFabricsQuestion', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NPFFabricsQuestion />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<NPFFabricsQuestion />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})