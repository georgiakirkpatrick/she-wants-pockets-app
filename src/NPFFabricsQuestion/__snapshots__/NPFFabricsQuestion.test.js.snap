// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFFabricsQuestion render the UI as expected 1`] = `
<div
  id="fabric-question"
>
  <section
    className="FormPage"
  >
    <h1>
      Fabrics
    </h1>
    <form>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          The primary fabric (or material) is the main fabric used in the garment or accessory.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          
        </p>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <p>
          The primary fabric (or material) is required.
        </p>
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              checked={false}
              id="primary-fabric"
              name="primary-fabric"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="primary-fabric"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Garments and accessories sometimes include secondary fabrics as trims or accents.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          
        </p>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <p>
          Does this product include a secondary fabric?
        </p>
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              checked={false}
              id="secondary-fabric"
              name="secondary-fabric"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="secondary-fabric"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          
        </p>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <p>
          Does this product include a lining fabric?
        </p>
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              checked={false}
              id="lining-fabric"
              name="lining-fabric"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="lining-fabric"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
    </form>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter_previous"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 NPFFooter_arrow"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter_next"
      onClick={[Function]}
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 NPFFooter_arrow"
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
